nmmo/task/predicate.py

```diff
@@ -0,0 +1,256 @@
+    assert len(agents) > 0, "Team must have at least one agent"
+    self.name = name if name else f"Agent{str(agents).replace(' ', '')}"
+    self._agents = agents
+
+  @property
+  def agents(self) -> List[int]:
+    return self._agents
+
+  def __contains__(self, ent_id: int):
+    return ent_id in self._agents
+
+  def __len__(self):
+    return len(self._agents)
+
+  def description(self) -> Dict:
+    return {
+      "type": "Group",
+      "name": self.name,
+      "agents": self._agents
+    }
+
+  def member(self, member):
+    assert member < len(self._agents)
+
+    if len(self._agents) == 1:
+      return self
+
+    # returning a group of one
+    return Group([self._agents[member]], f"{self.name}.{member}")
+
+
+# TODO(kywch): consider removing the tight coupling between policy - team
+#   Currently, this should produce the same map as the env realm spawn
+#   However, this may change later.
+class TeamHelper:
+  def __init__(self, agents: List[int], num_teams: int):
+    assert len(agents) % num_teams == 0
+    self.team_size = len(agents) // num_teams
+    self._team_to_ent, self._ent_to_team = self._map_ent_team(agents, num_teams)
+
+  def _map_ent_team(self, agents, num_teams):
+    _team_to_ent = {}
+    _ent_to_team = {}
+    for ent_id in agents:
+      # to assigne agent 1 to team 0, and so forth
+      pop_id = (ent_id - 1) % num_teams
+      _ent_to_team[ent_id] = pop_id
+      if pop_id in _team_to_ent:
+        _team_to_ent[pop_id].append(ent_id)
+      else:
+        _team_to_ent[pop_id] = [ent_id]
+
+    return _team_to_ent, _ent_to_team
+
+  def team(self, pop_id: int) -> Group:
+    assert pop_id in self._team_to_ent, "Wrong pop_id"
+    return Group(self._team_to_ent[pop_id], f"Team.{pop_id}")
+
+  def own_team(self, ent_id: int) -> Group:
+    assert ent_id in self._ent_to_team, "Wrong ent_id"
+    pop_id = self._ent_to_team[ent_id]
+    return Group(self._team_to_ent[pop_id], f"Team.{pop_id}")
+
+  def left_team(self, ent_id: int) -> Group:
+    assert ent_id in self._ent_to_team, "Wrong ent_id"
+    pop_id = (self._ent_to_team[ent_id] - 1) % len(self._team_to_ent)
+    return Group(self._team_to_ent[pop_id], f"Team.{pop_id}")
+
+  def right_team(self, ent_id: int) -> Group:
+    assert ent_id in self._ent_to_team, "Wrong ent_id"
+    pop_id = (self._ent_to_team[ent_id] + 1) % len(self._team_to_ent)
+    return Group(self._team_to_ent[pop_id], f"Team.{pop_id}")
+
+  def all(self) -> Group:
+    return Group(list(self._ent_to_team.keys()), "All")
+
+
+class Predicate:
+  # Predicate: states something about the subject
+  def __init__(self, subject:Group, *args):
+    assert len(subject) > 0, "There must be at least one agent in subject."
+    self._name = self._task_name(subject, args)
+    self._subject = subject.agents
+
+  @property
+  def name(self):
+    return self._name
+
+  @property
+  def subject(self) -> List[int]:
+    return self._subject
+
+  def _task_name(self, subject:Group, args):
```

Comment:
minor formatting, but you need a space after ": " here and in a bunch of places. If lint doesn't catch it, could you add a rule?

Reply:
Oh I didn't know that. I'm fixing it as I see it. 

I looked into pylint, and unfortunately, pylint removed `bad-whitespace` check from 2.6.0. 
https://pylint.readthedocs.io/en/latest/whatsnew/2/2.6/full.html#what-s-new-in-pylint-2-6-0

`bad-continuation and bad-whitespace have been removed, black or another formatter can help you with this better than Pylint`